tnc/modem.py

```diff
@@ -1354,22 +1360,23 @@ def calculate_fft(self) -> None:
                         # If we have a signal, increment our channel_busy delay counter
                         # so we have a smoother state toggle
                         if np.sum(slotdfft[slotdfft > avg + 15]) >= 200 and not TNC.transmitting:
-                            ModemParam.channel_busy = True
+                            addDelay=True
                             ModemParam.channel_busy_slot[slot] = True
-                            # Limit delay counter to a maximum of 200. The higher this value,
-                            # the longer we will wait until releasing state
-                            channel_busy_delay = min(channel_busy_delay + 10, 200)
                         else:
-                            # Decrement channel busy counter if no signal has been detected.
-                            channel_busy_delay = max(channel_busy_delay - 1, 0)
-                            # When our channel busy counter reaches 0, toggle state to False
-                            if channel_busy_delay == 0:
-                                ModemParam.channel_busy = False
-                                ModemParam.channel_busy_slot[slot] = False
-
+                            ModemParam.channel_busy_slot[slot] = False
                         # increment slot
                         slot += 1
-
+                    if (addDelay):
```

Comment:
This will possibly not be reset, because we don't have a "addDelay=False" . Question: Why is this in a separate if clause and not part of "if np.sum(slotdfft[slotdfft > avg + 15]) >= 200 and not" ? 

Is there are reason for putting "channel_busy_delay" as a global ( ModemParam.channel_busy_delay )?

Reply:
addDelay gets reset every time the while loop restarts.

It's outside of the for loop because during slot detection, it could get triggered for each slot on a single pass.  You could have a delay of 50 if each slot is busy on a single pass, but also the delay will be lessened quicker if each slot is not busy.

Reply:
The intent is to only increase delay if one or more slots are busy, or decrease delay if all slots are not busy.

Reply:
Ah now I got it, sorry, didnt see the entire code because of github